src/ray/raylet/node_manager.cc

```diff
@@ -337,6 +335,18 @@ void NodeManager::Heartbeat() {
     last_debug_dump_at_ms_ = now_ms;
   }
 
+  // Check worker heartbeat timeout times.
+  std::vector<std::shared_ptr<Worker>> dead_workers;
+  worker_pool_.GetDeadWorkers(RayConfig::instance().worker_heartbeat_timeout_times(),
+                              dead_workers);
+  if (!dead_workers.empty()) {
+    for (const auto &worker : dead_workers) {
+      RAY_LOG(INFO) << "Worker " << worker->GetWorkerId()
+                    << " dead because of timeout, pid: " << worker->Pid();
+      ProcessDisconnectClientMessage(worker->GetWorkerId(), worker->IsKilling());
```

Comment:
would `IsKilled` be better here?

Reply:
Use killing because we just sent the sigterm message, we only can confirm it when raylet server lost heartbeats.